extra-js/deno/autobuild/prepare

```diff
@@ -1,7 +1,24 @@
 export V8_FROM_SOURCE=1
 export CLANG_BASE_PATH='/usr/'
-export GN_ARGS="no_inline_line_tables=false use_lld=false is_clang=false \
+export GN_ARGS="no_inline_line_tables=false is_clang=false \
                 custom_toolchain=\"//build/toolchain/linux/unbundle:default\" host_toolchain=\"//build/toolchain/linux/unbundle:default\""
 export GN="/usr/bin/gn"
 export AR=ar
 export NM=nm
+
+# ld.gold is very likely to crash on amd64/aarch64/ppc64, and lld is unusable on loongson3
+if [[ "${CROSS:-$ARCH}" != 'loongson3' ]]; then
+   export GN_ARGS="$GN_ARGS use_lld=true"
+else
+   export GN_ARGS="$GN_ARGS use_lld=false use_gold=false"
+fi
+# workaround v8 issues
+abinfo "Fetching dependency sources ..."
+cargo fetch
+PATCHES_DIR="$SRCDIR/autobuild/patches"
+pushd /root/.cargo/registry/src/github.com-1ecc6299db9ec823/rusty_v8-0.15.0 || exit 1
```

Comment:
Uh okay this doesn't sound like a good idea...?

Reply:
Although Rust supports patching the dependencies, it requires a complete repository to do the trick, which is out of question in our case since the source to patch is V8 JS Engine.

Reply:
> Although Rust supports patching the dependencies, it requires a complete repository to do the trick, which is out of question in our case since the source to patch is V8 JS Engine.

I'd still prefer a dynamic reference though. Also I would suggest against using `pushd`. Try this:


```
(
    cd /path/to/source
    ...
)
```